src/pages/index.js

```diff
@@ -145,54 +144,135 @@ function openModal(modal) {
 }
 
 function closeModal(modal) {
-  modal.classList.remove("modal_opened");
+  modal.classList.remove(modalLikedCSSClass);
 
   modal.removeEventListener("click", handleModalOutsideClick);
   modal.removeEventListener("keydown", handleModalEscapeKey);
 }
 
+function addCardToCardList(card, method = "prepend") {
+  const cardElement = getCardElement(card);
+  cardsList[method](cardElement);
+}
+
+function handleModalOutsideClick(evt) {
+  if (evt.target.classList.contains(modalLikedCSSClass)) {
+    closeModal(evt.target);
+  }
+}
+
+function handleModalEscapeKey(evt) {
+  if (evt.key === "Escape") {
+    closeModal(evt.currentTarget);
+  }
+}
+
+function handleToggleLikeCard(evt, data) {
+  api
+    .changeCardLikeStatus({ id: data._id, isLiked: data.isLiked })
+    .then(() => {
+      evt.target.classList.toggle(cardLikedCSSClass);
+    })
+    .catch(console.error);
+}
+
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+
+  openModal(deleteCardModal);
+}
+
+//Form submit handlers
 function handleEditProfileSubmit(evt) {
   evt.preventDefault();
 
-  profileNameLabel.textContent = editProfileModalNameInput.value;
-  profileDescriptionParagraph.textContent =
-    editProfileModalDescriptionInput.value;
+  editProfileSubmitButton.textContent = savingSaveButtonText;
 
   disableButton(editProfileSubmitButton, validationConfig);
 
-  closeModal(editProfileModal);
+  api
+    .editUserInfo({
+      name: editProfileModalNameInput.value,
+      about: editProfileModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileNameLabel.textContent = data.name;
+      profileDescriptionParagraph.textContent = data.about;
+
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      editProfileSubmitButton.textContent = defaultSaveButtonText;
+    });
 }
 
-function handleNewPostSubmit(evt) {
+function handleEditProfileAvatarSubmit(evt) {
   evt.preventDefault();
 
-  const card = {
-    name: newPostModalCaptionInput.value,
-    link: newPostModalImageLinkInput.value,
-  };
+  editProfileAvatarSubmitButton.textContent = savingSaveButtonText;
+  console.log("tesT");
 
-  addCardToCardList(card);
+  api
+    .editUserAvatar({ avatar: editProfileAvatarModalImageInput.value })
+    .then((data) => {
+      profileAvatar.src = data.avatar;
 
-  evt.target.reset();
+      evt.target.reset();
 
-  closeModal(newPostModal);
+      editProfileAvatarSubmitButton.textContent = defaultSaveButtonText;
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.

Reply:
Ah good point, updated these lines to be in finally blocks.